test/powershell/engine/Basic/NativeCommandErrorHandling.Tests.ps1

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+# Functional tests to verify that errors (non-terminating and terminating) appropriately
+# when $PSNativeCommandUseErrorActionPreference is $true
+
+Describe 'Native command error handling tests' -Tags 'CI' {
+    Context 'PSNativeCommandUseErrorActionPreference is $true' {
```

Comment:
I'd combine tests for true and false (and maybe for $ErrorActionPreference values) and use TestCases.

Reply:
The only things I would add here are:

- Testing on the properties of the exception that we're supposed to implement, if they're available beyond the error message
- A test with a try/catch
- A test with a trap

Reply:
Adding these in my PR